specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json

```diff
@@ -0,0 +1,367 @@
+          {
+            "$ref": "#/parameters/LocationParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ComputeDiagnosticsList"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the status of a scheduled disk inspection request.": {
+            "$ref": "./examples/GetDiagnostics.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection": {
+      "get": {
+        "tags": [
+          "diagnostics_DiskInspection"
+        ],
+        "operationId": "DiskInspection_Get",
+        "description": "Gets a specific Comptue diagnostic for a subscription in a location.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ComputeDiagnosticBase"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the status of a scheduled disk inspection request.": {
+            "$ref": "./examples/GetDiskInspection.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ComputeDiagnosticsList": {
+      "description": "Lists all available Compute diagnostics for a subscription in a location.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ComputeDiagnosticBase"
+          }
+        }
+      }
+    },
+    "ComputeDiagnosticBase": {
+      "description": "Contains metadata of a diagnostic type",
+      "properties": {
+        "id": {
+          "description": "Identifier for the type of diagnostic",
+          "type": "string"
+        },
+        "name": {
+          "description": "Friendly name for the type of diagnostic",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/DiagnosticProperties"
+        }
+      }
+    },
+    "DiagnosticProperties": {
+      "description": "Contains additional properties of a diagnostic",
+      "properties": {
+        "supportedResourceTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Describes what are the supported resource types for a diagnostic",
+          "x-ms-examples": [ "VMs" ]
+        }
+      }
+    },
+    "RunDiskInspectionInput": {
+      "properties": {
+        "resourceGroup": {
+          "description": "Resource Group name.",
+          "type": "string"
+        },
+        "virtualMachineName": {
+          "description": "Virtual machine name.",
+          "type": "string"
```

Comment:
Use generic property name resourceId or resource name  so that you can extend disk inspection functionality to other resources in compute in future.